### PR TITLE
travis: Update travis to do a `make distcheck` instead of just `make check`

### PR DESCRIPTION
**Pull request by darconeous**
Doing a `make distcheck` instead of just a `make check` ensures that the build system is sane.

I fully expect this commit to break travis, so expect a few more commits to come in on top of this commit correcting any issues that crop up.

This issue is related to #40.

### CODE REVIEW

**Comment by jwhui** [comment]
Should we fix `make distcheck` before merging this?

File: third_party/nlbuild-autotools/repo/scripts/mkversion [hunk 1]
```diff
@@ -161,7 +161,11 @@ gitversion() {
 				fi
 
 			else
-				gitversion=`${PRINTF} "g${head}"`
```
Comment (mszczodrak) [inline third_party/nlbuild-autotools/repo/scripts/mkversion]: this should go to third_party project commit directly
OT just pulls in third party code from their master repos

**Comment by darconeous** [comment]
Yes, don't merge until travis says it is ok!

File: third_party/nlbuild-autotools/repo/scripts/mkversion (hunk 1 above)
Comment (darconeous) [inline third_party/nlbuild-autotools/repo/scripts/mkversion]: Turned out I was falling down the rabbit hole. The most recent version makes no changes to nlbuild-autotools.

**Comment by darconeous** [comment]
Woo-hoo, all set!

File: Makefile.am [hunk 2]
```diff
@@ -91,6 +93,14 @@ OPENTHREAD_VERSION                ?= $(shell cat $(VERSION_FILE) 2> /dev/null)
 PACKAGE_VERSION                    = $(OPENTHREAD_VERSION)
 VERSION                            = $(PACKAGE_VERSION)
 
+distdir = $(PACKAGE)-$(shell \
+if [ "$(origin OPENTHREAD_VERSION)" != "file" ]; then \
+    echo "$(OPENTHREAD_VERSION)" ;            \
+else                                                             \
+    $(abs_top_nlbuild_autotools_dir)/scripts/mkversion           \
+        -b "$(OPENTHREAD_VERSION)" "$(top_srcdir)";    \
```
Comment (jwhui) [inline Makefile.am]: @gerickson code style has the '\' aligned.

File: examples/cli/Makefile.am [hunk 3]
```diff
@@ -40,10 +40,10 @@ soc_CPPFLAGS                                                   = \
     $(NULL)
 
 soc_LDADD                                                      = \
-    $(top_srcdir)/src/core/libopenthread.a                       \
-    $(top_srcdir)/src/cli/libopenthread-cli.a                    \
-    $(top_srcdir)/examples/platform/posix/libopenthread-posix.a  \
-    $(top_srcdir)/third_party/mbedtls/libmbedcrypto.a            \
+    $(top_builddir)/src/core/libopenthread.a                     \
+    $(top_builddir)/src/cli/libopenthread-cli.a                  \
+    $(top_builddir)/examples/platform/posix/libopenthread-posix.a \
```
Comment (jwhui) [inline examples/cli/Makefile.am]: @gerickson code style has the '\' aligned.

File: tests/unit/Makefile.am [hunk 4]
```diff
@@ -54,9 +54,9 @@ AM_CPPFLAGS                                                    = \
     $(NULL)
 
 COMMON_LDADD                                                   = \
-    $(top_srcdir)/src/core/libopenthread.a                       \
-    $(top_srcdir)/examples/platform/posix/libopenthread-posix.a  \
-    $(top_srcdir)/third_party/mbedtls/libmbedcrypto.a            \
+    $(top_builddir)/src/core/libopenthread.a                     \
+    $(top_builddir)/examples/platform/posix/libopenthread-posix.a \
```
Comment (jwhui) [inline tests/unit/Makefile.am]: @gerickson code style has the '\' aligned.

**Comment by jwhui** [comment]
Looks great!  Thanks for doing this.  Some style nit comments inline.

File: examples/cli/Makefile.am (hunk 3 above)
Comment (darconeous) [inline examples/cli/Makefile.am]: Is there a line width we are trying to keep the slashes at, or is it just at the furthest out that keeps them all inline?

**Comment by darconeous** [comment]
Will fix the nits tomorrow.

File: examples/cli/Makefile.am (hunk 3 above)
Comment (jwhui) [inline examples/cli/Makefile.am]: I believe it's furthest out that keeps them all inline.  @gerickson , do you have any opinion?

Comment (gerickson) [inline examples/cli/Makefile.am]: On May 18, 2016, at 11:04 PM, Jonathan Hui <notifications@github.com> wrote:
> In examples/cli/Makefile.am <https://github.com/openthread/openthread/pull/46#discussion_r63826346>:
> 
> > @@ -40,10 +40,10 @@ soc_CPPFLAGS                                                   = \
> >      $(NULL)
> >  
> >  soc_LDADD                                                      = \
> > -    $(top_srcdir)/src/core/libopenthread.a                       \
> > -    $(top_srcdir)/src/cli/libopenthread-cli.a                    \
> > -    $(top_srcdir)/examples/platform/posix/libopenthread-posix.a  \
> > -    $(top_srcdir)/third_party/mbedtls/libmbedcrypto.a            \
> > +    $(top_builddir)/src/core/libopenthread.a                     \
> > +    $(top_builddir)/src/cli/libopenthread-cli.a                  \
> > +    $(top_builddir)/examples/platform/posix/libopenthread-posix.a \
> I believe it's furthest out that keeps them all inline. @gerickson <https://github.com/gerickson> , do you have any opinion?
> 
Furthest out.

Best,

Grant

**Comment by darconeous** [comment]
Fixed style nits.

**Pull request merged**